### ISSUE TITLE
Support custom .obsidian config directory

### ISSUE THREAD

**Issue by r4zzz4k** [jamiebrynes7/obsidian-todoist-plugin#160]
**Describe the bug**
Recent version of Obsidian supports changing config directory from `.obsidian` to some custom value. For example, this is useful if one wants to have different plugin set for desktop and mobile apps, or in case synchronization software doesn't support hidden directories at all. Currently, the plugin searches for the token file in `.obsidian`. It should access new [`Vault.configDir`](https://github.com/obsidianmd/obsidian-api/blob/9bb30633586c6649474b2affdf5b87efdd86498f/obsidian.d.ts#L3331-L3336) field if it's present, falling back to default `.obsidian` value if that's not present.

**To Reproduce**
1. Open Obsidian
2. Copy your `.obsidian` directory as `.obsidian.desktop`.
3. Open Settings
4. Open 'About' screen
5. Put `.obsidian.desktop` in "Override config folder".
6. Close Obsidian
7. Delete `.obsidian` directory.
8. Open Obsidian again.
Note that the old directory is back again, the plugin created an empty file for the token. It also asks to enter the token again.

**Expected behavior**
Plugin should search for the token file in new config directory.

**Comment by megclaypool** [comment]
Whoops -- I accidentally just filed a duplicate bug report in [#170](https://github.com/jamiebrynes7/obsidian-todoist-plugin/issues/170)! I'll just copy over the only bit not in this issue:

In obsidian-todoist-plugin/src/token.ts you are hard-coding the location of the todoist api to the .obsidian folder:

```js
export function getTokenPath(): string {
  return `.obsidian/todoist-token`;
}
```

**Comment by r4zzz4k** [comment]
Implemented by @jamiebrynes7 in #168. Thank you!
